documentation/modules/oauth/con-oauth-authentication-client-options.adoc

```diff
@@ -41,39 +44,74 @@ Here you can see the communication flows, for different configurations of Kafka
-. Kafka client session is established if the token is valid.
+. The Kafka client authenticates with the Kafka broker using the SASL OAUTHBEARER mechanism to pass the long-lived access token.
+. The Kafka broker validates the access token by calling a token introspection endpoint on the authorization server, using its own client ID and secret.
+. A Kafka client session is established if the token is valid.
 
 [id='oauth-token-jwt-{context}']
 .Client using long-lived access token, with broker performing fast local validation
 
 image:oauth-jwt-signature-token.png[Client using long-lived access token with broker performing fast local validation]
 
-. Kafka client authenticates with the Kafka broker using the _SASL OAUTHBEARER_ mechanism to pass the long-lived access token.
-. Kafka broker validates the access token locally using JWT token signature check, and local token introspection.
+. The Kafka client authenticates with the Kafka broker using the SASL OAUTHBEARER mechanism to pass the long-lived access token.
+. The Kafka broker validates the access token locally using a JWT token signature check and local token introspection.
 
 WARNING: Fast local JWT token signature validation is suitable only for short-lived tokens as there is no check with the authorization server if a token has been revoked.
 Token expiration is written into the token, but revocation can happen at any time, so cannot be accounted for without contacting the authorization server.
 Any issued token would be considered valid until it expires.
+
+== Example client authentication flows using the SASL PLAIN mechanism
+
+You can use the following communication flows for Kafka authentication using the OAuth PLAIN mechanism.
+
+* xref:oauth-plain-client-id-{context}[Client using a client ID and secret, with the broker obtaining the access token for the client]
+* xref:oauth-plain-access-token-{context}[Client using a long-lived access token without a client ID and secret]
+
+[id='oauth-plain-client-id-{context}']
+.Client using a client ID and secret, with the broker obtaining the access token for the client
+
+image:oauth-plain-client-id.png[Client using a client ID and secret, with the broker obtaining the access token for the client]
+
+. The Kafka client passes a `clientId` as a username and a `secret` as a password.
+. The Kafka broker uses a token endpoint to pass the `clientId` and `secret` to the authorization server.
+. The authorization server returns a fresh access token in exchange or an error if the client credentials aren't valid.
+. The Kafka broker validates the token in one of the following ways:
+.. If a token introspection endpoint is specified, the Kafka broker validates the access token by calling the endpoint on the authorization server.
+A session is established if the token validation was successful.
+.. If local token introspection is used, there is no request made to the authorization server.
```

Comment:
I think a possible rewrite. 

If local token introspection is used, a request is not made to the authorization server.